File: 406-queue-reconstruction-by-height/leochang/406.cpp
class Solution {
public:
	vector<pair<int, int>> reconstructQueue(vector<pair<int, int>>& people) {
		sort(people.begin(), people.end(), comp);
		vector<pair<int, int>> ret;
		for (auto &p: people) {
			ret.insert(ret.begin() + p.second, p);
		}
		return ret;
	}
	static bool comp(pair<int, int>& p1, pair<int, int>& p2) {
		if (p1.first != p2.first) {
			return p1.first > p2.first;
		} else {
			return p1.second < p2.second;
		}
	}
};

File: 406-queue-reconstruction-by-height/ken/406.py
class Solution(object):
    def reconstructQueue(self, people):
        """
        :type people: List[List[int]]
        :rtype: List[List[int]]
        """

        queue = []
        people.sort(key = lambda x : (-x[0], x[1]))

        for i in xrange(len(people)):
            height = people[i][0]
            cnt = people[i][1]
            idx = 0

            for j in xrange(len(queue)):
                if cnt == 0:
                    break
                cnt -= 1
                idx += 1

            queue.insert(idx, people[i])

        return queue
